test/plugin-e2e-specs.js

```diff
@@ -146,4 +147,80 @@ describe('FakePlugin', function () {
       });
     });
   }
+
+  describe('cli args handling for plugin args', function () {
+    let server = null;
+    before(async function () {
+      // then start server if we need to
+      const args = {port: TEST_PORT, host: TEST_HOST, appiumHome, plugins: ['fake'], pluginArgs: FAKE_PLUGIN_ARGS};
+      server = await appiumServer(args);
+    });
+    after(async function () {
+      if (server) {
+        await server.close();
+      }
+    });
+
+    it('should recieve user cli args for plugin if passed in', async function () {
+      const driver = await wdio(wdOpts);
+      const {sessionId} = driver;
+
+      try {
+        const {data} = await axios.get(`${baseUrl}/${sessionId}/fakecliargs`);
```

Comment:
now that I think about it I'm a little concerned that both the driver and the plugin establish the same new GET path. would be nice if they were a bit different so that we were sure in a scenario like this we are getting the plugin's CLI args and not fakedriver's